quicksettings/QuickSettings.hx

```diff
@@ -1,189 +1,343 @@
    */
   public function setKey(char:Int):Void;
-  
+
   /**
    * Set the position of the QuickSettings panel
    */
   public function setPosition(x:Float, y:Float):Void;
-  
+
   /**
    * Set the size of the QuickSettings panel
    */
   public function setSize(w:Float, h:Float):Void;
-  
+
   /**
    * Set the width of the QuickSettings panel
    */
   public function setWidth(w:Float):Void;
-  
+
   /**
    * Set the height of the QuickSettings panel
    */
   public function setHeight(h:Float):Void;
-  
+
+  // If the panel is draggable, you can have it snap to a grid when dropped. And you can specify the size of that grid:
+  public function setSnapToGrid(bool:Bool):Void;
+  public function setGridSize(number:Int):Void;
+
   /**
    * Global change handler. This callback will be called whenever any change is made to any control in this panel.
    */
-  public function setGlobalChangeHandler(callback:Dynamic->Dynamic):Void;
-  
+  public function setGlobalChangeHandler(callback:Dynamic->Void):Void;
+
   public function bindRange<T:Float>(property:String, min:T, max:T, value:T, step:Float, object:Dynamic):Void;
   public function bindColor(property:String, color:String, object:Dynamic):Void;
   public function bindBoolean(property:String, value:Bool, object:Dynamic):Void;
   public function bindText(property:String, text:String, object:Dynamic):Void;
   public function bindDropDown<T>(property:String, items:Array<T>, object:Dynamic):Void;
+  public function bindDate(title:String, date:String, object:Dynamic):Void;
+  public function bindNumber(title:String, value:String, object:Dynamic):Void;
+  public function bindPassword(title:String, text:String, object:Dynamic):Void;
+  public function bindTextArea(title:String, text:String, object:Dynamic):Void;
+  public function bindTime(title:String, time:String, object:Dynamic):Void;
+
+}
+
+typedef OnFileChosen = {
+  var type : String;
+  var size : String;
+  var name : String;
+  var lastModifiedDate : String;
 }
+
+// not sure how usefull...
+
+typedef QSettings =
+{
+  @optional var title : String;               // optional string, default "QuickSettings"
+  @optional var x : Int;                      // optional number, default 0
+  @optional var y : Int;                      // optional number, default 0
+  @optional var draggable : Bool;             // optional bool,   default true
+  @optional var collapsible : Bool;           // optional bool,   default true
+  @optional var snapToGrid : Bool;            // optional bool,   default false
+  @optional var gridSize : Int;               // optional number  default 0
+  @optional var controls : Array<QControls>;   // optional array of control objects
+};
+
+typedef QControls =
+{
+  var type : String;                          // required string
+  var title : String;                         // required string
+  @optional var value : Dynamic;              // optional value:
```

Comment:
I think it is better to use a block comment `/** comment **/` before each var for nicer code hints.